servers/dsub/jobs/controllers/jobs_controller.py

```diff
@@ -175,6 +175,16 @@ def _generate_dstat_jobs(provider, query, create_time_max=None,
     if not create_time_max and query.end:
         create_time_max = query.end
 
+    # If the query explicitly requests submitted jobs which aren't running
```

Comment:
opt: I'm not fully sure  what the impact of this would be on the rest of the code, but an alternative approach to this might be to switch this generator to yielding converted common Jobs API jobs; then after converting in this function you could do a simpler check

```
if query.statuses and api_job.status not in query.statuses:
  continue
```

If it would cause issues with the job ID token creation, for example, then I wouldn't bother.

Reply:
Good suggestion, ended up taking some minor refactoring but it ends up making things much cleaner working with a job model rather than a dict.